src/commands/synthetics/utils/public.ts

```diff
@@ -453,21 +453,41 @@ export const getReporter = (reporters: Reporter[]): MainReporter => ({
 export const getTestAndOverrideConfig = async (
   api: APIHelper,
   // TODO SYNTH-12989: Clean up deprecated `config` in favor of `testOverrides`
-  {config, testOverrides, id, suite}: TriggerConfig,
+  triggerConfig: TriggerConfig,
   reporter: MainReporter,
   summary: InitialSummary,
   isTunnelEnabled?: boolean
 ): Promise<TestNotFound | TestSkipped | TestWithOverride> => {
-  const normalizedId = normalizePublicId(id)
+  // TODO SYNTH-12989: Clean up deprecated `config` in favor of `testOverrides`
+  const testOverrides = replaceConfigWithTestOverrides(triggerConfig.config, triggerConfig.testOverrides)
 
-  if (!normalizedId) {
-    throw new CriticalError('INVALID_CONFIG', `No valid public ID found in: \`${id}\``)
+  if (isLocalTriggerConfig(triggerConfig)) {
+    if (!triggerConfig.id) {
+      throw new CriticalError(
+        'INVALID_CONFIG',
+        'Linking local test definitions to an existing remote test is currently mandatory.'
+      )
+    }
```

Comment:
What would be the consequences of removing this?
I know that we don't support LTD without a public id for now, but it might be easier for us to keep this enforced only in the backend, so there are no `datadog-ci` version with this requirement.
Can we keep track of the expected results just with their result id?

Reply:
There are 3 places that require the public ID: starting tunnels, and both reporters (default and JUnit). For now, I'm replacing the public ID with `local`, so that we still have something to show. We will iterate on that